### PR TITLE
[2.13.x] DDF-3893 The IdP should account for the NameIDPolicy in AuthnRequests

### PR DESCRIPTION
**Pull request by bakejeyner**
##### ABBREVIATED REVIEW BETWEEN 2.13.X AND MASTER IS IN EFFECT
[Master PR](https://github.com/codice/ddf/pull/3412)

#### Who is reviewing it? 
@blen-desta 
@brjeter 

#### Ask 2 committers to review/merge the PR and tag them here.
@brendan-hofmann
@clockard
@coyotesqrl

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: platform/security/idp/security-idp-plugin/security-idp-plugin-nameidpolicy/pom.xml [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <parent>
+    <groupId>ddf.security.idp</groupId>
+    <artifactId>security-idp-plugin</artifactId>
+    <version>2.14.0-SNAPSHOT</version>
```
Comment (brjeter) [inline platform/security/idp/security-idp-plugin/security-idp-plugin-nameidpolicy/pom.xml]: ❗️ This version needs to be updated

**Comment by coyotesqrl** [comment]
build now

**Comment by brjeter** [comment]
build now (flaky NoClassDefFound failure)

**Comment by coyotesqrl** [comment]
Skipped Sonarqube for abbreviated review.

**Pull request merged**